pandas/core/indexes/base.py

```diff
@@ -4333,6 +4333,11 @@ def equals(self, other):
             # if other is not object, use other's logic for coercion
             return other.equals(self)
 
+        if isinstance(other, ABCMultiIndex):
+            if not is_object_dtype(self):  # d-level MultiIndex can equal d-tuple Index
```

Comment:
```suggestion
            if not is_object_dtype(self.dtype):  # d-level MultiIndex can equal d-tuple Index
```

Can you verify that this case is actually tested?

Reply:
This case is covered by at least two existing tests:

 - `TestIndex.test_equals` in pandas/tests/indexes/test_base.py with a MultiIndex `indices`
 - `test_equals` in pandas/tests/indexes/multi/test_equivalence.py with a MultiIndex `idx`

According to the doc for `is_object_dtype`, it can be used to "Check whether an **array-like** or dtype is of the object dtype" (emphasis mine). Is there a reason to prefer calling it directly on the dtype rather than the Index?

Reply:
Thanks.

Using the dtype rather than the array can be faster.